### ISSUE TITLE
Provide component status in devworkspace-client

### ISSUE THREAD

**Issue by benoitf** [eclipse-che/che#19151]
### Is your enhancement related to a problem? Please describe.
Add component status (like endpoint URL merged into devfile components structure) + flag to know if component has been added by the tooling or if it was defined by user in its devfile (for example I have a component like 'java' with container including java/maven, etc ==> this is a user component, while che-theia if I didn't explicitely set an editor --> tooling component

### Describe the solution you'd like
Provide status by grabbing information in devWorkspace routings, workspace pods and devWorkspace objects instead of letting clients pull information themselves and aggregate data on their own

### Describe alternatives you've considered
Should be done until https://github.com/devfile/devworkspace-operator/issues/261 is exposed

**Comment by sleshchenko** [comment]
I think https://github.com/devfile/devworkspace-operator/issues/261 should be way to go, we just need to agree on the model for needed info
